### PR TITLE
Feat/zked

### PR DESCRIPTION
**Pull request by mikiquantum**
Signature changes + regenerated contracts

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -32,7 +32,8 @@ required = ["github.com/centrifuge/centrifuge-ethereum-contracts", "github.com/r
 
 [[override]]
   name = "github.com/centrifuge/centrifuge-ethereum-contracts"
-  revision = "863c5efc0a49aff1e9c1cf2ef5f72f62c65425ed"
+  source = "github.com/mikiquantum/centrifuge-ethereum-contracts"
```
Comment (mikiquantum) [inline Gopkg.toml]: Will change once the contracts PR is merged.

File: Makefile [hunk 2]
```diff
@@ -81,6 +81,7 @@ gen-abi-bindings: ## Generates GO ABI Bindings
 gen-abi-bindings: install-deps abigen-install
 	npm install --prefix vendor/github.com/centrifuge/centrifuge-ethereum-contracts
 	npm run compile --prefix vendor/github.com/centrifuge/centrifuge-ethereum-contracts
+	@mkdir -p tmp/contracts
```
Comment (mikiquantum) [inline Makefile]: Bug.

File: documents/coredocument_test.go [hunk 3]
```diff
@@ -838,3 +862,246 @@ func TestCoreDocument_UpdateAttributes_both_nil(t *testing.T) {
 	assert.Len(t, upattrs, 0)
 	assert.Len(t, uattrs, 0)
 }
+
+func TestValidateFullNFTProof(t *testing.T) {
```
Comment (mikiquantum) [inline documents/coredocument_test.go]: Added unit test to make sure that we always validate NFT proofs properly.

File: documents/processor.go [hunk 4]
```diff
@@ -105,12 +105,12 @@ func (dp defaultProcessor) PrepareForSignatureRequests(ctx context.Context, mode
...
 
 	// calculate the document data root
-	sr, err := model.CalculateDocumentDataRoot()
+	ddr, err := model.CalculateDocumentDataRoot()
 	if err != nil {
 		return errors.New("failed to calculate document data root: %v", err)
 	}
 
-	sig, err := self.SignMsg(sr)
+	sig, err := self.SignMsg(append(ddr, []byte{0}...))
```
Comment (mikiquantum) [inline documents/processor.go]: Maybe add a standard function for document consensus signatures here?

File: documents/service.go [hunk 5]
```diff
@@ -196,10 +196,17 @@ func (s service) RequestDocumentSignature(ctx context.Context, model Model, coll
 
 	srvLog.Infof("document received %x with document data root %x", model.ID(), sr)
 
-	sig, err := acc.SignMsg(sr)
+	var transitionFlag int8
```
Comment (vedhavyas) [inline documents/service.go]: we can make this as byte so that we avoid conversion later ?

**Comment by vedhavyas** [comment]
@mikiquantum any chance we increase the test coverage for this?

**Pull request merged**